SmartDeviceLinkTests/DevAPISpecs/SDLTextAndGraphicUpdateOperationSpec.m

```diff
@@ -797,6 +797,47 @@
         });
     });
 
+    // updating with error state
+    describe(@"updating with error state", ^{
+        beforeEach(^{
+            updatedState = [[SDLTextAndGraphicState alloc] init];
+            updatedState.textField1 = field1String;
+            updatedState.primaryGraphic = testArtwork;
+            updatedState.secondaryGraphic = testArtwork2;
+
+            testOp = [[SDLTextAndGraphicUpdateOperation alloc] initWithConnectionManager:testConnectionManager fileManager:mockFileManager currentCapabilities:windowCapability currentScreenData:emptyCurrentData newState:updatedState currentScreenDataUpdatedHandler:^(SDLTextAndGraphicState * _Nullable newScreenData, NSError * _Nullable error) {} updateCompletionHandler:nil];
+            [testOp start];
+        });
+
+        it(@"should reset to current screen data for equivalent properties in updated state and error state", ^{
+            // Create an error state that matches the updated state, which should reset the updated state
+            SDLTextAndGraphicState *errorState = [[SDLTextAndGraphicState alloc] init];
+            errorState.textField1 = updatedState.textField1;
+            errorState.primaryGraphic = updatedState.primaryGraphic;
+            errorState.secondaryGraphic = updatedState.secondaryGraphic;
+
+            [testOp updateTargetStateWithErrorState:errorState];
+
+            expect(updatedState.textField1).to(beNil());
+            expect(updatedState.primaryGraphic).to(beNil());
+            expect(updatedState.secondaryGraphic).to(beNil());
+        });
+
+        it(@"should not reset to current screen data for non equivalent properties in updated state and error state", ^{
+            // Create an error state that does not match the updated state, which should not reset the updated state
+            SDLTextAndGraphicState *errorState = [[SDLTextAndGraphicState alloc] init];
+            errorState.textField1 = nil;
+            errorState.primaryGraphic = nil;
+            errorState.secondaryGraphic = nil;
+
+            [testOp updateTargetStateWithErrorState:errorState];
+
+            expect(updatedState.textField1).toNot(beNil());
+            expect(updatedState.primaryGraphic).toNot(beNil());
+            expect(updatedState.secondaryGraphic).toNot(beNil());
+        });
```

Comment:
These will need to check all of the data, not just selected items. Additionally, the format of the test should be all "updated state" data is full, all "current screen data" is `nil`. Then send an error state with data that matches updated state, and now everything should be empty. Then send an error state with data that doesn't match the updated state, and everything should still match its original state. Then send an error state with everything nil, and it should still match its original state. LMK if you have questions. We should want to make sure we're covering everything here.